Metadata/MetadataConverter.php

```diff
@@ -80,11 +80,15 @@ public function convert(ClassHierarchyMetadata $metadata)
             $definition->setProperties($classMetadata->properties);
 
             if (null !== $classMetadata->decorates) {
+                if (!isset($classMetadata->decorationInnerName) && isset($classMetadata->decoration_inner_name)) {
```

Comment:
It should fail when the property will be removed, it will permit to not forget to remove this check.